lib/mix/tasks/appsignal.diagnose.ex

```diff
@@ -139,20 +139,15 @@ defmodule Mix.Tasks.Appsignal.Diagnose do
     download_parsing_error = Map.has_key?(report, "download_parsing_error")
     install_parsing_error = Map.has_key?(report, "installation_parsing_error")
 
-    cond do
-      download_parsing_error ->
-        do_print_parsing_error("download", report)
-
-      install_parsing_error ->
-        do_print_download_report(report)
-
-      true ->
-        nil
+    if download_parsing_error && install_parsing_error do
+      do_print_parsing_error("download", report)
+      do_print_parsing_error("installation", report)
     end
 
-    if install_parsing_error, do: do_print_parsing_error("installation", report)
-
-    if !download_parsing_error && !install_parsing_error do
+    if install_parsing_error do
+      do_print_download_report(report)
+      do_print_parsing_error("installation", report)
+    else
       do_print_installation_report(report)
     end
```

Comment:
Not sure if it'd be clearer, but another way to write this would be using a `case`:

```elixir
case {download_parsing_error, install_parsing_error} do
  {true, true} ->   # first error case
  {false, true} ->  # second error case
  {false, false} -> # success!
end
```